README.md

```diff
@@ -1,2 +1,283 @@
+
+The identifier is fully stateless and will remain the same after reinstalling or clearing application data.
+
+# Table of Contents
+1. [Quick start](#quick-start)
+2. [API Reference](#api-reference)
+3. [Initialization and usage](#initialization-and-usage)
+4. [Advanced usage](#advanced-usage)
+
+
+## Quick start
+
+### Install as Gradle dependency
+
+Add these lines to `build.gradle` file of your project.
+
+
+```gradle
+allprojects {
+	repositories {
+		...
+		maven { url 'https://jitpack.io' }	
+	}
+}
+```
+
+### Add the dependency to your project
+
+Add these lines to `build.gradle` file of your module.
+This library depends on `kotlin-stdlib`. If your application is written in Java, add `kotlin-stdlib` dependency first (it's lightweight and has excellent backward and forward compatibility).
+
+```gradle
+dependencies {
+	implementation 'com.github.fingerprintjs:fingerprint-android:1.0.0'
+	
+	// Add this line for java-project
+	implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"	
+}
+```
+
+## API Reference
+
+### Device ID and Fingerprint
+
+The library operates with two entities. 
+
+1. `deviceId` - it is an ID of the device, based on system-provided IDs. Currently, it is based on Google Service Framework ID, and ANDROID_ID if the first is unavailable. It's strict, which means the uniqueness of the ID for every device. Also, it keeps the same after application reinstalling.
+2. `fingerprint` - it is a digital fingerprint of a device. Basically, it is a hash, calculated from a lot of available platform signals. `fingerprint` **is not a strict ID**, so there is a probability that two different devices will have the same fingerprint. Also, there is a probability, that the same device will have a different fingerprint in different moments due to system update, settings changing, etc.
+
+#### Why do we need them both?
+It is a good question, most of use-cases can be covered with the `deviceId`. But the `GSF_ID` and `ANDROID_ID`can be spoofed much easier, than device data such as CPU info or sensors list. 
+
+Another reason for using `fingerprint` is a possible restriction of access to `deviceId` due to privacy policy changes trend.
+
+
+
+### Initialization and usage
+
+Initialize the library and get `fingerprint` and `deviceId`.
+
+In Kotlin
+
+```kotlin
+
+// Initialization
+val fingerprinter =
+            FingerprinterFactory.getInitializedInstance(applicationContext)
+
+// Simple usage
+val fingerprint = fingerprinter.fingerprint()
+val deviceId = fingerprinter.deviceId()
+
+```
+
+In Java
+
+```java
+
+// Initialization
+Fingerprinter fingerprinter =
+                FingerprinterFactory.getInitializedInstance(getApplicationContext());
+                
+// Simple usage
+String fingerprint = fingerprinter.fingerprint();
+String deviceId = fingerprinter.deviceId();
+
+```
+### Advanced usage
+The additional entities, provided by the library are `BaseFingerprinter` and `Configuration`.
+
+#### BaseFingerprinter
```

Comment:
why is it called `BaseFingerprinter`? This looks more like a base signal provider.
A fingerprinter is an entity that takes signal providers (one or several of them), queries signals from providers and generates an identifier. 
Please let me know if my understanding is incorrect here.